Fix typo in join vignette: Replace "banana" with "soda" (#6798) (#6801)
Rdatatable/data.table: vignettes/datatable-joins.Rmd

  - It didn't add the prefix `i.` to any column.
  
- **Row level**

>   - All rows from in the `i` table were kept as we never received any soda but row is still part of the results.

Reviewer: ```suggestion
  - All rows from the `i` table were kept: the soda entry from `Products` that was not matched by any row in `ProductReceived` is still part of the results.
```

- Finding the nearest match.
- Comparing ranges of values between tables.

> ### Key Constraints
> - **Column Requirements**:
>   - Only column *names* (not expressions) can be used with binary operators
>   - Left operand must be from `x` (left table)
>   - Right operand must be from `i` (right table)
> - **Naming Behavior**:
>   - Joined columns inherit names from `x` but contain values from `i`
> 

Reviewer: This is not requested in the linked issue, nor described+motivated in the PR, so I'm dropping it. I have some quibbles with whether it should be here / it needs some refinement. Better to just merge the fix as is for now.

```suggestion
```
